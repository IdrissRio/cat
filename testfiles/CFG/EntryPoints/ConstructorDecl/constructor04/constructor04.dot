digraph G {
graph [splines=ortho, nodesep="1", ranksep="1"]
node [shape=rect, fontname=Arial];
{ rank = same; "Program[0]"}
{ rank = same; "List[1]"}
{ rank = same; "CompilationUnit[2]"}
{ rank = same; "List[3]"; "List[4]"}
{ rank = same; "ClassDecl[5]"}
{ rank = same; "Modifiers[22]"; "Opt[25]"; "List[26]"; "List[27]"; "InstanceInitialization[6]"; "InstanceInitialization[14]"}
{ rank = same; "List[23]"; "ConstructorDecl[28]"; "ConstructorDecl[36]"; "Entry[7]"; "Exit[8]"; "Entry[15]"; "Exit[16]"}
{ rank = same; "Modifier[24]"; "Entry[29]"; "Implicit[29]"; "Modifiers[29]"; "List[31]"; "List[32]"; "Opt[33]"; "Block[34]"; "Exit[29]"; "ExprStmt[36]"; "Entry[37]"; "Implicit[37]"; "Modifiers[37]"; "List[39]"; "List[44]"; "Opt[45]"; "Block[46]"; "Exit[37]"; "ExprStmt[48]"}
{ rank = same; "List[30]"; "List[35]"; "SuperConstructorAccess[37]"; "List[38]"; "ParameterDeclaration[40]"; "List[47]"; "SuperConstructorAccess[49]"}
{ rank = same; "List[38]"; "Modifiers[41]"; "PrimitiveTypeAccess[43]"; "List[50]"}
{ rank = same; "List[42]"}
"Program[0]" [label="Program\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[1]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"CompilationUnit[2]" [label="CompilationUnit\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[3]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[4]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ClassDecl[5]" [label="ClassDecl\n < constructor04 >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[22]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[23]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifier[24]" [label="Modifier\n < public >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[25]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[26]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[27]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ConstructorDecl[28]" [label="ConstructorDecl\n < constructor04 >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Entry[29]" [label="Entry\n ", fillcolor=white   style=filled  ]
"Implicit[29]" [label="Implicit\n < Instance >", fillcolor=white   style=filled  ]
"Modifiers[29]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[30]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[31]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[32]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[33]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Block[34]" [label="Block\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[35]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Exit[29]" [label="Exit\n ", fillcolor=white   style=filled  ]
"ExprStmt[36]" [label="ExprStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"SuperConstructorAccess[37]" [label="SuperConstructorAccess\n < super >", fillcolor=white   style=filled  ]
"List[38]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ConstructorDecl[36]" [label="ConstructorDecl\n < constructor04 >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Entry[37]" [label="Entry\n ", fillcolor=white   style=filled  ]
"Implicit[37]" [label="Implicit\n < Instance >", fillcolor=white   style=filled  ]
"Modifiers[37]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[38]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[39]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ParameterDeclaration[40]" [label="ParameterDeclaration\n < t >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[41]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[42]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[43]" [label="PrimitiveTypeAccess\n < int >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[44]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[45]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Block[46]" [label="Block\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[47]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Exit[37]" [label="Exit\n ", fillcolor=white   style=filled  ]
"ExprStmt[48]" [label="ExprStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"SuperConstructorAccess[49]" [label="SuperConstructorAccess\n < super >", fillcolor=white   style=filled  ]
"List[50]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"InstanceInitialization[6]" [label="InstanceInitialization\n < Instance >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Entry[7]" [label="Entry\n ", fillcolor=white   style=filled  ]
"Exit[8]" [label="Exit\n ", fillcolor=white   style=filled  ]
"InstanceInitialization[14]" [label="InstanceInitialization\n < Static >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Entry[15]" [label="Entry\n ", fillcolor=white   style=filled  ]
"Exit[16]" [label="Exit\n ", fillcolor=white   style=filled  ]
"CompilationUnit[2]" -> "List[3]" [style=dashed, color=gray]
"List[23]" -> "Modifier[24]" [style=dashed, color=gray]
"Modifiers[22]" -> "List[23]" [style=dashed, color=gray]
"ClassDecl[5]" -> "Modifiers[22]" [style=dashed, color=gray]
"ClassDecl[5]" -> "Opt[25]" [style=dashed, color=gray]
"ClassDecl[5]" -> "List[26]" [style=dashed, color=gray]
"ConstructorDecl[28]" -> "Entry[29]" [style=dashed, color=gray]
"ConstructorDecl[28]" -> "Implicit[29]" [style=dashed, color=gray]
"Modifiers[29]" -> "List[30]" [style=dashed, color=gray]
"ConstructorDecl[28]" -> "Modifiers[29]" [style=dashed, color=gray]
"ConstructorDecl[28]" -> "List[31]" [style=dashed, color=gray]
"ConstructorDecl[28]" -> "List[32]" [style=dashed, color=gray]
"ConstructorDecl[28]" -> "Opt[33]" [style=dashed, color=gray]
"Block[34]" -> "List[35]" [style=dashed, color=gray]
"ConstructorDecl[28]" -> "Block[34]" [style=dashed, color=gray]
"ConstructorDecl[28]" -> "Exit[29]" [style=dashed, color=gray]
"SuperConstructorAccess[37]" -> "List[38]" [style=dashed, color=gray]
"ExprStmt[36]" -> "SuperConstructorAccess[37]" [style=dashed, color=gray]
"ConstructorDecl[28]" -> "ExprStmt[36]" [style=dashed, color=gray]
"List[27]" -> "ConstructorDecl[28]" [style=dashed, color=gray]
"ConstructorDecl[36]" -> "Entry[37]" [style=dashed, color=gray]
"ConstructorDecl[36]" -> "Implicit[37]" [style=dashed, color=gray]
"Modifiers[37]" -> "List[38]" [style=dashed, color=gray]
"ConstructorDecl[36]" -> "Modifiers[37]" [style=dashed, color=gray]
"Modifiers[41]" -> "List[42]" [style=dashed, color=gray]
"ParameterDeclaration[40]" -> "Modifiers[41]" [style=dashed, color=gray]
"ParameterDeclaration[40]" -> "PrimitiveTypeAccess[43]" [style=dashed, color=gray]
"List[39]" -> "ParameterDeclaration[40]" [style=dashed, color=gray]
"ConstructorDecl[36]" -> "List[39]" [style=dashed, color=gray]
"ConstructorDecl[36]" -> "List[44]" [style=dashed, color=gray]
"ConstructorDecl[36]" -> "Opt[45]" [style=dashed, color=gray]
"Block[46]" -> "List[47]" [style=dashed, color=gray]
"ConstructorDecl[36]" -> "Block[46]" [style=dashed, color=gray]
"ConstructorDecl[36]" -> "Exit[37]" [style=dashed, color=gray]
"SuperConstructorAccess[49]" -> "List[50]" [style=dashed, color=gray]
"ExprStmt[48]" -> "SuperConstructorAccess[49]" [style=dashed, color=gray]
"ConstructorDecl[36]" -> "ExprStmt[48]" [style=dashed, color=gray]
"List[27]" -> "ConstructorDecl[36]" [style=dashed, color=gray]
"ClassDecl[5]" -> "List[27]" [style=dashed, color=gray]
"InstanceInitialization[6]" -> "Entry[7]" [style=dashed, color=gray]
"InstanceInitialization[6]" -> "Exit[8]" [style=dashed, color=gray]
"InstanceInitialization[14]" -> "Entry[15]" [style=dashed, color=gray]
"InstanceInitialization[14]" -> "Exit[16]" [style=dashed, color=gray]
"ClassDecl[5]" -> "InstanceInitialization[6]" [style=dashed, color=gray]
"ClassDecl[5]" -> "InstanceInitialization[14]" [style=dashed, color=gray]
"List[4]" -> "ClassDecl[5]" [style=dashed, color=gray]
"CompilationUnit[2]" -> "List[4]" [style=dashed, color=gray]
"List[1]" -> "CompilationUnit[2]" [style=dashed, color=gray]
"Program[0]" -> "List[1]" [style=dashed, color=gray]
"Exit[16]" -> "Entry[15]" [color=red, constraint=false]
"Entry[15]" -> "Exit[16]" [color=blue, constraint=false, xlabel=" "] 
"Exit[8]" -> "Entry[7]" [color=red, constraint=false]
"Entry[7]" -> "Exit[8]" [color=blue, constraint=false, xlabel=" "] 
"Exit[29]" -> "Implicit[29]" [color=red, constraint=false]
"Implicit[29]" -> "Exit[29]" [color=blue, constraint=false, xlabel=" "] 
"Implicit[29]" -> "SuperConstructorAccess[37]" [color=red, constraint=false]
"SuperConstructorAccess[37]" -> "Implicit[29]" [color=blue, constraint=false, xlabel=" "] 
"SuperConstructorAccess[37]" -> "Entry[29]" [color=red, constraint=false]
"Entry[29]" -> "SuperConstructorAccess[37]" [color=blue, constraint=false, xlabel=" "] 
"Exit[37]" -> "Implicit[37]" [color=red, constraint=false]
"Implicit[37]" -> "Exit[37]" [color=blue, constraint=false, xlabel=" "] 
"Implicit[37]" -> "SuperConstructorAccess[49]" [color=red, constraint=false]
"SuperConstructorAccess[49]" -> "Implicit[37]" [color=blue, constraint=false, xlabel=" "] 
"SuperConstructorAccess[49]" -> "Entry[37]" [color=red, constraint=false]
"Entry[37]" -> "SuperConstructorAccess[49]" [color=blue, constraint=false, xlabel=" "] 
"List[3]" -> "List[4]" [style=invis]
"InstanceInitialization[14]" -> "InstanceInitialization[6]" -> "Modifiers[22]" -> "Opt[25]" -> "List[26]" -> "List[27]" [style=invis]
"ConstructorDecl[28]" -> "ConstructorDecl[36]" [style=invis]
"Entry[29]" -> "Modifiers[29]" -> "List[31]" -> "List[32]" -> "Opt[33]" -> "Block[34]" -> "Exit[29]" [style=invis]
"Entry[37]" -> "Modifiers[37]" -> "List[39]" -> "List[44]" -> "Opt[45]" -> "Block[46]" -> "Exit[37]" [style=invis]
"Modifiers[41]" -> "PrimitiveTypeAccess[43]" [style=invis]
"Entry[7]" -> "Exit[8]" [style=invis]
"Entry[15]" -> "Exit[16]" [style=invis]

subgraph cluster_legend {
{node [style=filled, fillcolor=1, colorscheme="pastel13"]
Nodes [label="Total Nr. Nodes
50", fillcolor=1]
Edges [label="Total Nr. Edges
65", fillcolor=1]
NodesCFG [label="CFG Nr. Nodes
9", fillcolor=3]
EdgesCFG [label="CFG Nr. Edges
8", fillcolor=3]
Nodes-> "Program[0]" -> NodesCFG  [style="invis"]
Edges -> "Program[0]" ->EdgesCFG [style="invis"]}
label = "Statistics of the framework: NEXTFramework"
style="solid"
ranksep=0.05
nodesep=0.01
labelloc = b
len=2
}}