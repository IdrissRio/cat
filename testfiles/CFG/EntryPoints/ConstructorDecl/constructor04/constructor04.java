public class constructor04 {
  constructor04() {}
  constructor04(int t) {}
}