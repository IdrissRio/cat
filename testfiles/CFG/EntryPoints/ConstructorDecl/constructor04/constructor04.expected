NodeName: Entry[15] Successors: [Exit[16]]
NodeName: Exit[16] Successors: []
NodeName: Entry[7] Successors: [Exit[8]]
NodeName: Exit[8] Successors: []
NodeName: Entry[29] Successors: [SuperConstructorAccess[37]]
NodeName: SuperConstructorAccess[37] Successors: [Implicit[29]]
NodeName: Implicit[29] Successors: [Exit[29]]
NodeName: Exit[29] Successors: []
NodeName: Entry[37] Successors: [SuperConstructorAccess[49]]
NodeName: SuperConstructorAccess[49] Successors: [Implicit[37]]
NodeName: Implicit[37] Successors: [Exit[37]]
NodeName: Exit[37] Successors: []

NodeName: Exit[16] Predecessor: [Entry[15]]
NodeName: Entry[15] Predecessor: []
NodeName: Exit[8] Predecessor: [Entry[7]]
NodeName: Entry[7] Predecessor: []
NodeName: Exit[29] Predecessor: [Implicit[29]]
NodeName: Implicit[29] Predecessor: [SuperConstructorAccess[37]]
NodeName: SuperConstructorAccess[37] Predecessor: [Entry[29]]
NodeName: Entry[29] Predecessor: []
NodeName: Exit[37] Predecessor: [Implicit[37]]
NodeName: Implicit[37] Predecessor: [SuperConstructorAccess[49]]
NodeName: SuperConstructorAccess[49] Predecessor: [Entry[37]]
NodeName: Entry[37] Predecessor: []
